content/master/guides/composition-functions.md

```diff
@@ -0,0 +1,1002 @@
+---
+title: Composition Functions
+state: alpha
+---
+
+Composition Functions allow you to supplement or replace your Compositions with
+advanced logic. You can build a Function using general purpose programming
+languages such as Go or Python, or relevant tools such as Helm, kustomize, or
+CUE. Functions compliment contemporary "Patch and Transform" (P&T) style
+Composition - it's possible to use only P&T, only Functions, or a mix of both in
+the same Composition.
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: example
+spec:
+  compositeTypeRef:
+    apiVersion: database.example.org/v1alpha1
+    kind: XPostgreSQLInstance
+  functions:
+  - name: my-cool-Function
+    type: Container
+    container:
+      image: xpkg.upbound.io/my-cool-Function:0.1.0
+```
+
+A Composition Function is a short-lived OCI container that tells Crossplane how
+to reconcile a Composite Resource (XR). The preceding example shows a minimal
+`Composition` that uses a Composition Function. Note that it has a `functions`
+array rather than the typical P&T style array of `resources`.
+
+## Enabling Functions
+
+To enable support for Composition Functions you must:
+
+ * Enable the alpha feature flag in Crossplane.
+ * Deploy a runner that is responsible for running Functions.
```

Comment:
the main reason is that it's easy to omit a word when reading. If you read "do not" you can omit "not". If you read "don't" you won't omit a word.

https://developers.google.com/style/contractions
https://docs.gitlab.com/ee/development/documentation/styleguide/#contractions

Reply:
Interestingly the GitLab docs seem to contradict the Google docs (at least, they specifically call out that you should use "do not" rather than "don't").

I've updated this page to use contractions everywhere I could, but I still feel that _requiring them everywhere_ is a bit heavy handed and would prefer to recommend them.

Reply:
My preference is contractions. We are still building the Crossplane style guide, but I feel strongly about contractions.